client/src/components/SessionForm.jsx

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from 'react';
+import { Modal, Form, Input, TextArea, Radio } from 'semantic-ui-react';
+
+import 'semantic-ui-css/semantic.min.css';
+import { addSession } from '../utils/apiWrapper';
+import '../css/SessionForm.scss';
+
+function SessionForm(props) {
+  const { button } = props;
+  const [open, setOpen] = useState(false);
+  const [isLater, setIsLater] = useState(false);
+  const [courseCode, setCourseCode] = useState('');
+  const [courseNumber, setCourseNumber] = useState();
+  const [courseSuffix, setCourseSuffix] = useState('');
+  const [location, setLocation] = useState('');
+  const [attendees, setAttendees] = useState('');
+  const [notes, setNotes] = useState('');
+  const [date, setDate] = useState();
+  const [startTime, setStartTime] = useState();
+  const [endTime, setEndTime] = useState();
+  const [endTimeDefined, setEndTimeDefined] = useState(false);
+
+  const processFormAndSubmit = async () => {
```

Comment:
A next step (no need to add this to this PR): add validation/typechecking. For example, making sure the start and end dates are valid and in the future. Another thing is making sure the required input fields are filled in on submit (so checking that the user did input a location, which is a required field)